### PR TITLE
Remove SocketInputStream (#753).

### PR DESCRIPTION
**Pull request by cesarbs**
#753

cc @halter73 @mikeharder

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/Http/Connection.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public class Connection : ConnectionContext, IConnectionControl
         private ConnectionFilterContext _filterContext;
         private LibuvStream _libuvStream;
         private FilteredStreamAdapter _filteredStreamAdapter;
-        private Task _readInputContinuation;
+        private Task _readInputTask;
```
Comment (muratg) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/Connection.cs]: Why rename?  Just curious...

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/Connection.cs]: Having "continuation" is the name is needlessly leaking information about what's happening inside `ReadInput()`.

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/Connection.cs]: A call to `ContinueWith()` that is.

**Comment by halter73** [comment]
Makes you wonder why it wasn't written this way in the first place. I promise there was a reason!

Basically before SslStream was really working, @lodejard and I were hoping that SslStream would smartly override CopyToAsync to call SocketInputStream with its own byte buffers to avoid the double buffer problem we now have. (Notice we now have to use a MemoryPoolBlock in FilterInputAsync to copy data from a byte buffer SslStream already allocated).

Those supposed benefits are completely worthless with the default Stream implementation of CopyToAsync though.

:shipit:

**Pull request merged**